functions/v1/upsert_user.sql

```diff
@@ -0,0 +1,42 @@
+SET search_path = v1;
+
+CREATE OR REPLACE FUNCTION upsert_user(
+  in_username      TEXT,
+  in_user_type     entity_type,
+  in_external_id   TEXT,
+  in_display_name  TEXT,
+  in_email_address TEXT
+) RETURNS TIMESTAMP WITH TIME ZONE
```

Comment:
This proc will not work if `in_user_type` is `internal` since it always results in a `NULL` password.  BTW, this is more of a statement than an indication of a problem ;)